src/data/docsTree.ts

```diff
@@ -0,0 +1,49 @@
+import type { DocsMetadata, DocsNode, DocsTree } from "$data/docs";
+
+const PATH_TRIM = /(?:..\/routes\/docs)((\/[\w-]+)+)\//;
+
+const getPages = () => {
+    console.log("SEARH")
+    console.trace();
+    const rawPages = import.meta.glob<DocsMetadata>(
+        ["../routes/docs/**/*/+page.md", "../routes/docs/+page.md"], // import all docs pages + base page
+        { eager: true, import: "metadata" }
+    );
+
+    return Object.entries(rawPages).map(([path, node]) => {
+        return {
+            title: node.title,
+            path: path.match(PATH_TRIM)?.[1] ?? ""
+        } as DocsNode;
+    });
+};
+
+const getTree = (pages: DocsNode[]) => {
+    const rawCategories = import.meta.glob<DocsMetadata>("../routes/docs/**/*/category.md", {
+        eager: true,
+        import: "metadata"
+    });
+
+    const categories: DocsTree = Object.entries(rawCategories).map(
+        ([path, node]) => {
+            const docsPath =
+                path.match(PATH_TRIM)?.[1] ?? "this is an invalid path";
```

Comment:
It's better to throw an error here instead of moving forward with potentially corrupt data.
```suggestion
                path.match(PATH_TRIM)?.[1] ?? throw Error(`this is an invalid docs page path: $path`);
```

Reply:
ok